app/src/main/java/com/navapp/navigation/route/RouteHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.navapp.navigation.route;
+
+import com.navapp.navigation.destination.Destination;
+import com.navapp.navigation.destination.data.Address;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+public class RouteHandler {
```

Comment:
What is even the purpose of this class? what does it do?

Reply:
There is already a comment discussing it.